src/geometry/LineUtil.js

```diff
@@ -257,10 +259,21 @@ export function polylineCenter(latlngs, crs) {
 		latlngs = latlngs[0];
 	}
 
+	var centroidLatLng = toLatLng([0, 0]);
+
+	var bounds = toLatLngBounds(latlngs);
+	var areaBounds = bounds.getNorthWest().distanceTo(bounds.getSouthWest()) * bounds.getNorthEast().distanceTo(bounds.getNorthWest());
+	// tests showed that below 1700 rounding errors are happening
+	if (areaBounds < 1700) {

```

Comment:
I have the suspicion that the threshold is not based on the area, but rather on the (maximum) magnitude of *either* the horizontal or vertical component.

Did these tests include very slim (e.g. small `x` but large `y`) polygons near e.g. latitude 0, longitude 179? Longitude 0 latitude 82?

Reply:
> I have the suspicion that the threshold is not based on the area, but rather on the (maximum) magnitude of either the horizontal or vertical component.

@IvanSanchez I do not really understand what this means, but every help is appriciated. This is error is much over my knowledge and this workaround was the best what I was eable to achieve. 

- Demo (v1.9.3) where you can edit a small polygon and see the center "jumping" https://jsfiddle.net/3m0d4Lqc/
- small `x` but large `y`: https://jsfiddle.net/p8wt2q3o/2/:
  - 1.9.3: 
  https://jsfiddle.net/p8wt2q3o/2/
![grafik](https://user-images.githubusercontent.com/19800037/218528200-967a5892-4edb-4547-87b7-2caa6a64c767.png)
  - new: 
![grafik](https://user-images.githubusercontent.com/19800037/218528298-26ec3020-5257-4f99-8d4e-105b3189e749.png)
- large`x` but small  `y`: https://jsfiddle.net/p8wt2q3o/2/:
  - 1.9.3: 
  https://jsfiddle.net/c4Lj2wup/
![grafik](https://user-images.githubusercontent.com/19800037/218529373-c8c04671-3665-4ea2-9203-01d4cbb2c9ef.png)
  - new: 
![grafik](https://user-images.githubusercontent.com/19800037/218529195-f73eed92-9c6f-43f3-b4c8-f18d9cab8c2c.png)

Reply:
This is due to how floating point arithmetic works. One of the diagrams in the [wikipedia page about floating point](https://en.wikipedia.org/wiki/Floating-point_arithmetic) sums it nicely:

![image](https://user-images.githubusercontent.com/1125786/220114492-1ee6f6b4-fea3-4965-bf2e-36f86f7d3c7a.png)

Precision is good near zero, but becomes worse the further from zero you are.

This can be seen in your jsfiddle. This is how it looks with a rectangle like `[[0.0445, 179.925448],[0.0447, 179.925449]]`:
![image](https://user-images.githubusercontent.com/1125786/220115030-8d599eb8-cadd-4753-b07d-6b794b4d137e.png)

...and see how it's different from a rectangle like `[[0.0445, 0.925448],[0.0447, 0.925449]]`:

![image](https://user-images.githubusercontent.com/1125786/220115424-3f8d6029-8dc5-4844-a237-ab030cc71598.png)

Which means that your code covers the worst case for bound latitude-longitude (-90,90; -180,180), which is good. But I can *guess* it will fail for maps with `L.CRS.Simple`, which can use larger coordinates.

I'm OK with that, but want you to be aware.